iml-gui/crate/src/components/command_modal.rs

```diff
@@ -3,27 +3,68 @@
 // license that can be found in the LICENSE file.
 
 use crate::{
-    components::{font_awesome, modal},
+    components::{font_awesome::*, modal},
+    dependency_tree::{build_direct_dag, DependencyDAG, Deps, Rich},
     extensions::{MergeAttrs as _, NodeExt as _, RequestExt as _},
     generated::css_classes::C,
     key_codes, sleep_with_handle, GMsg,
 };
 use futures::channel::oneshot;
-use iml_wire_types::{ApiList, Command, EndpointName};
+
+use iml_wire_types::{ApiList, Command, EndpointName, Job, Step};
+use regex::{Captures, Regex};
 use seed::{prelude::*, *};
-use std::{collections::HashSet, sync::Arc, time::Duration};
+use serde::de::DeserializeOwned;
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::{self, Display},
+    sync::Arc,
+    time::Duration,
+};
 
 /// The component polls `/api/command/` endpoint and this constant defines how often it does.
 const POLL_INTERVAL: Duration = Duration::from_millis(1000);
 
-#[derive(Default, Debug)]
-pub struct Model {
-    cancel: Option<oneshot::Sender<()>>,
-    loading: bool,
-    opens: HashSet<u32>,
-    pub commands_all: Vec<Arc<Command>>,
-    pub commands: Vec<Arc<Command>>,
-    pub modal: modal::Model,
+type Job0 = Job<Option<()>>;
```

Comment:
```suggestion
type Job0 = Job<Option<serde_json::Value>>;
```

This can't be hardcoded to unit, args coming back can be anything, and can vary between / within a collection.

Reply:
I see, thanks